backend/src/components/sagas.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+const config = require('../config/index');
+const {
+  logApiError, getBackendToken, getData, putData
+} = require('./utils');
+const HttpStatus = require('http-status-codes');
+const { FILTER_OPERATION, VALUE_TYPE, CONDITION } = require('../util/constants');
+const { SAGA_TYPES } = require('../util/constants/sagaTypes');
+
+async function getSagas(req, res) {
+  try {
+  let searchListCriteria = [];
+
+  if(req.query.searchCriteriaList) {
+    let searchQueries = JSON.parse(req.query.searchCriteriaList);
+
+    Object.keys(searchQueries).forEach(element => {
+      if(!searchQueries[element] || (Array.isArray(searchQueries[element]) && Object.keys(searchQueries[element]).length === 0)) return; //do not add to search if null value
+      let operation = FILTER_OPERATION.IN;
+      let valueType = VALUE_TYPE.STRING;
+
+      switch (element) {
+        case 'createDate':
+        case 'updateDate':
+          if(!searchQueries[element].from) return;
+          if (!searchQueries[element].end) {
+            searchQueries[element].end = searchQueries[element].from;
+          }
+          searchQueries[element].end = searchQueries[element].end.replace(/\//g, '-');
+          searchQueries[element].from = searchQueries[element].from.replace(/\//g, '-');
+          searchQueries[element] = searchQueries[element].from + 'T00:00:00,' + searchQueries[element].end + 'T23:59:59';
+
+          operation = FILTER_OPERATION.BETWEEN;
+          valueType = VALUE_TYPE.DATE_TIME;
+          break;
+        case 'retryCount':
+          operation = FILTER_OPERATION.GREATER_THAN;
+          valueType = VALUE_TYPE.INTEGER;
+          searchQueries[element] = 0;
+          break;
+        case 'sagaId':
+          operation = FILTER_OPERATION.EQUAL;
+          valueType = VALUE_TYPE.UUID;
+          break;
+        case 'status':
+        case 'sagaName':
+        case 'sagaState':
+        default:
+          if(Object.keys(searchQueries[element]).length === 0) return;
+          searchQueries[element] = searchQueries[element].join(',');
+          break;
+      }
+      searchListCriteria.push({key: element, condition: CONDITION.AND, operation: operation, value: searchQueries[element], valueType: valueType});
+    });
+  }
+  const search = [
+    {
+      searchCriteriaList: searchListCriteria
+    }
+  ];
+    const params = {
+      params: {
+        pageNumber: req?.query?.pageNumber,
+        pageSize: req?.query?.pageSize || 10,
+        sort: req?.query?.sort,
+        searchCriteriaList: JSON.stringify(search)
+      }
+    };
+
+    let url;
+    switch (req?.query?.sagaType) {
+      case SAGA_TYPES.PEN_REQUEST_BATCH:
```

Comment:
Would it be better to define an object which maps sagaType to url? so the switch statement would not be needed and if new saga type is added, the function will not be changed. I think this approach would make the codebase more maintainable.